src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -44,6 +70,38 @@ class HMPPSAuthService(
     return if (serviceProviderOrgs.isEmpty()) null else serviceProviderOrgs[0]
   }
 
+  fun getUserDetail(user: AuthUser): UserDetail {
+    return if (user.authSource == "auth") {
+      val url = UriComponentsBuilder.fromPath(authUserDetailLocation).buildAndExpand(user.userName).toString()
+      hmppsAuthApiWebClient.get().uri(url)
+        .retrieve()
+        .bodyToMono(AuthUserDetailResponse::class.java)
+        .map {
+          if (!it.verified) {
+            throw UnverifiedEmailException()
+          }
+          UserDetail(it.firstName, it.email)
+        }
+        .block()
+    } else {
+      val detailUrl = UriComponentsBuilder.fromPath(userDetailLocation).buildAndExpand(user.userName).toString()
+      val emailUrl = UriComponentsBuilder.fromPath(userEmailLocation).buildAndExpand(user.userName).toString()
+      Mono.zip(
+        hmppsAuthApiWebClient.get().uri(detailUrl)
+          .retrieve()
+          .bodyToMono(UserDetailResponse::class.java)
+          .map { it.name.substringBefore(' ') },
```

Comment:
this is a sucky way to get the firstname, but i don't know what else to do

Reply:
could we poll the user details from its source when we first validate a token like this?
i.e. `AuthUser` &rarr; pull details from source? `userDetailsFromAuth`, `userDetailsFromDelius`?

Reply:
at this point we have no token for the user or anything like it. this user was passed to us in the body of the 'assign' request. so we just have username, userid and authsource. we could use community api to get the user details, but i'm not sure it's worth it. the first name is only used in the email to say "hi {firstName}", so it's not super important.

Reply:
a downside of doing this is that community api doesn't store email address for user. so to get the information we require to send the email we would have to make a call to hmpps auth for the email and delius for the name. which is added complexity.